lib/awsutil/api.go

```diff
@@ -17,6 +17,17 @@ func CreateSession(accountProfileName string) (*session.Session, error) {
 		SharedConfigState: session.SharedConfigEnable})
 }
 
+// CredentialsOptions contains options for loading credentials
+type CredentialsOptions struct {
+	// The path of the credentials file.
+	// If empty, defaults to ~/.aws/credentials
+	CredentialsPath string
+
+	// The path of the config file.
+	// If empty, defaults to ~/.aws/config
+	ConfigPath string
+}
+
```

Comment:
This isn't what I had in mind. I was thinking of adding flags in this package. That way the application doesn't need to be changed in order to take advantage of this improvement. See what the `lib/log/serverlogger` package does.

I also note that you're not supporting the `AWS_CONFIG` environment variable. Again, that should be handled here so that the application doesn't have to be changed in order to support it. The code should use the values from command-line flags, and the default values for those flags should come from the environment (first `AWS_CONFIG` if set, else the existing default).

Reply:
Thanks for looking at this code.

With this code I have written it will be trivial to have the existing methods, LoadCredentials() and TryLoadCredentials() support the AWS_CONFIG environment variable and command line flags. Did you have a preference for what the command line flags should be called?

While I agree that what you ask for will be useful, I have a special use case where I need to specify the location of the config file from within my program rather than relying on the user to specify command line flags. I am building an AWS config file by using the organizations service. The trouble is that not all the role accounts I get using that service are properly set up. To winnow out the accounts that aren't properly set up, I write a trail config file out to the /tmp directory. Next I use TryLoadCredentialsWithOptions() to load credentials using that config file I just wrote. Finally, I use the results of that call to filter out the role accounts that didn't load and I remove the config file in the tmp directory. Note that in my use case, the user running my program can't and shouldn't know in advance the location of this trial config file in the /tmp directory as I am using the ioutil package to create the temporary file.

Reply:
What I am proposing is this in order of precedence:

1. Use values program passes in for config file (highest precedence) overrides everything (my use case)
2. Use command line options
3. Use AWS_CONFIG environment variable (lowest precedence).

Admittedly my use case is a very special one, but if I use the AWS_CONFIG environment variable to specify the location of the temporary config file from within my program, which seems hacky to me, the user running my program could break it by specifying command line options.

Reply:
OK, your use-case makes sense. For the name of the command-line flags, perhaps: `-awsConfigFile` and `-awsCredentialsFile`. I'm open to other suggestions, though.